clients/algoliasearch-client-kotlin/client/src/commonMain/kotlin/com/algolia/client/extensions/SearchClient.kt

```diff
@@ -534,3 +541,39 @@ public suspend fun SearchClient.indexExists(indexName: String): Boolean {
 
   return true
 }
+
+public data class SearchDisjunctiveFacetingResponse(
+  val response: SearchResponse,
+  val disjunctiveFacets: Map<String, Map<String, Int>>,
+)
+
+/**
+ * Method used for perform search with disjunctive facets.
+ *
+ * @param indexName The name of the index in which the search queries should be performed
+ * @param searchParamsObject The search query params.
+ * @param refinements Refinements to apply to the search in form of dictionary with
+ *  facet attribute as a key and a list of facet values for the designated attribute.
+ *  Any facet in this list not present in the `disjunctiveFacets` set will be filtered conjunctively (with AND operator).
+ * @param disjunctiveFacets Set of facets attributes applied disjunctively (with OR operator)
+ * @param requestOptions Configure request locally with RequestOptions.
+ * @return SearchDisjunctiveFacetingResponse - a struct containing the merge response from all the
+ * disjunctive faceting search queries, and a list of disjunctive facets
+ * @throws NoSuchElementException if there are no search response from the multi-query search
+ */
+public suspend fun SearchClient.searchDisjunctiveFaceting(
+  indexName: String,
+  searchParamsObject: SearchParamsObject,
```

Comment:
```suggestion
  searchParams: SearchParamsObject,
```

Is it the same in swift? Does the user needs to know the type?

Reply:
I disagree with this one: I'm following the gen code pattern
I think it will be more consistent for the user to have a single way of naming variables across our clients
And yes, we want to enforce a searchParamsObject, and not a string

Reply:
I mean generally they are not aware of the multiple shapes etc. for them it's just search params, but having object is not a huge deal either it still make sense in most languages